providers/inwx/inwxProvider.go

```diff
@@ -250,13 +254,18 @@ func (api *inwxAPI) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Cor
 	return corrections, nil
 }
 
-// GetNameservers returns the default nameservers for INWX.
-func (api *inwxAPI) GetNameservers(domain string) ([]*models.Nameserver, error) {
+// getDefaultNameservers returns back string map with default nameservers  based on e.g. sandbox mode.
+func (api *inwxAPI) getDefaultNameservers() []string {
 	if api.sandbox {
-		return models.ToNameservers(InwxSandboxDefaultNs)
+		return InwxSandboxDefaultNs
```

Comment:
I originally had the code like this as well, but apparently golint prefers to not have the `else` clause if the `if` clause already contains a `return` statement :man_shrugging:

Reply:
tbh, I don't understand why `golint` should complain about that. It works just fine :D

Reply:
Lint complains because it creates a situation where code is never executed.

```
func (api *inwxAPI) getDefaultNameservers() []string {
  if api.sandbox {
    return InwxSandboxDefaultNs
  } else {
    return InwxProductionDefaultNs
  }
  // Suppose the next developer adds this line:
  fmt.Printf("I can't figure out why this never executes!")
}
```

Reply:
Ah, I see! Any suggestion how to solve it somehow else, or should we leave it as-is?

Reply:
The Go style is, when possible, return errors early:

```go
func (api *inwxAPI) getDefaultNameservers() []string {
  if validate something is bad {
    return fmt.Errorf(...)
  }
  if more input validation {
    return fmt.Errorf(...)
  }
  // now code can assume clean input, which makes it easier to write code
  x := api.foo
}
```

Consider this code which is anti the style guide. It creates many levels of indentation which is hard to read and difficult to maintain.  The error message is far away from the code that found the error.  Also, adding new validation checks cause you to re-indent your code. Sure, gofmt does that for you, but the diff will be huge.

```go
func (api *inwxAPI) getDefaultNameservers() []string {
  if validate something is ok  {
    if validate something else is ok {
      if validate another thing {
        // all input is good. now I can write code here
        // that is clean.
      } else {
        return error from "another thing"
    } else {
      return error from "something else"
  } else {
    return error from "something is ok"
  }
}
```

Reply:
So basically:
```go
func (api *inwxAPI) getDefaultNameservers() []string {
	if api.sandbox {
		return InwxSandboxDefaultNs
	}
	return InwxProductionDefaultNs
}
```
That's what I've committed now.

Reply:
Great!